app/shared/Shared/Services/AzureDocumentService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Azure;
+using Azure.Search.Documents;
+using Azure.Search.Documents.Models;
+using Shared.Models;
+
+public interface IDocumentService
+{
+    Task<SupportingContentRecord[]> QueryDocumentsAsync(
+               string? query = null,
+               float[]? embedding = null,
+               RequestOverrides? overrides = null,
+               CancellationToken cancellationToken = default);
```

Comment:
I'd recommend instead having an overload to make it clearer that you need either the `query` or `embedding`.

```suggestion
    Task<SupportingContentRecord[]> QueryDocumentsAsync(
        string? query = null,
        RequestOverrides? overrides = null,
        CancellationToken cancellationToken = default);
        
    Task<SupportingContentRecord[]> QueryDocumentsAsync(
        float[]? embedding = null,
        RequestOverrides? overrides = null,
        CancellationToken cancellationToken = default);
```

Reply:
Thanks for suggestion, SearchClient supports hybrid mode though, that's why this API accepts both embedding and query

https://learn.microsoft.com/en-us/azure/search/hybrid-search-overview